doc/modules/calibration.rst

```diff
@@ -111,30 +123,70 @@ predicted probabilities of the `k` estimators in the
 `calibrated_classifiers_` list. The output of :term:`predict` is the class
 that has the highest probability.
 
-The regressor that is used for calibration depends on the `method`
-parameter. `'sigmoid'` corresponds to a parametric approach based on Platt's
-logistic model [3]_, i.e. :math:`p(y_i = 1 | f_i)` is modeled as
-:math:`\sigma(A f_i + B)` where :math:`\sigma` is the logistic function, and
-:math:`A` and :math:`B` are real numbers to be determined when fitting the
-regressor via maximum likelihood. `'isotonic'` will instead fit a
-non-parametric isotonic regressor, which outputs a step-wise non-decreasing
-function (see :mod:`sklearn.isotonic`).
-
-An already fitted classifier can be calibrated by setting `cv="prefit"`. In
-this case, the data is only used to fit the regressor. It is up to the user
+Alternatively an already fitted classifier can be calibrated by setting
+`cv="prefit"`. In this case, the data is not split and all of it is used to
+fit the regressor. It is up to the user
 make sure that the data used for fitting the classifier is disjoint from the
 data used for fitting the regressor.
 
-:class:`CalibratedClassifierCV` can calibrate probabilities in a multiclass
-setting if the base estimator supports multiclass predictions. The classifier
-is calibrated first for each class separately in a one-vs-rest fashion [4]_.
-When predicting probabilities, the calibrated probabilities for each class
+:class:`CalibratedClassifierCV` supports the use of two 'calibration'
+regressors: 'sigmoid' and 'isotonic'. Both these regressors only
+support 1-dimensional data (e.g., binary classification output) but are
+extended for multiclass classification if the `base_estimator` supports
+multiclass predictions. For multiclass predictions,
+:class:`CalibratedClassifierCV` calibrates for
+each class separately in a :ref:`ovr_classification` fashion [4]_. When
+predicting
+probabilities, the calibrated probabilities for each class
 are predicted separately. As those probabilities do not necessarily sum to
 one, a postprocessing is performed to normalize them.
 
 The :func:`sklearn.metrics.brier_score_loss` may be used to evaluate how
 well a classifier is calibrated.
 
+Sigmoid
+^^^^^^^
+
+The sigmoid regressor is based on Platt's logistic model [3]_:
+
+.. math::
+       p(y_i = 1 | f_i) = \frac{1}{1 + \exp(A f_i + B)}
+
+where :math:`y_i` is the true label of sample :math:`i` and :math:`f_i`
+is the output of the un-calibrated classifier for sample :math:`i`. :math:`A`
+and :math:`B` are real numbers to be determined when fitting the regressor via
+maximum likelihood.
+
+The sigmoid method is biased in that it assumes the :ref:`calibration curve
+<calibration_curve>` of the un-calibrated model has a sigmoid shape and is
+symmetrical [1]_. It is thus most effective when the un-calibrated model is
+under-confident and has similar errors for both high and low
```

Comment:
my understanding is that we assume over-confidence for low probabilities and under-confidence for high probabilities (which are then compensated by the shape of the logistic function)?

also by "similar errors" do we mean errors with similar absolute values / magnitude?

Reply:
Oh I am confused. I thought that a 'sigmoid' shape of calibration curve meant under-confident classifier and 'transposed-sigmoid' shape meant over-confident model. This is just from the example: https://scikit-learn.org/stable/auto_examples/calibration/plot_calibration_curve.html#sphx-glr-auto-examples-calibration-plot-calibration-curve-py

I think similar errors is in terms of the shape of the calibration curve - the sigmoid is symmetrical in shape. Since we are dealing with difference in predicted probability and frequency of true positives per bin, i would say similar absolute difference?

Reply:
> my understanding is that we assume over-confidence for low probabilities and under-confidence for high probabilities (which are then compensated by the shape of the logistic function)?

Ok I see how I was wrong here

> Oh I am confused. I thought that a 'sigmoid' shape of calibration curve meant under-confident classifier and 'transposed-sigmoid' shape meant over-confident model. This is just from the example: scikit-learn.org/stable/auto_examples/calibration/plot_calibration_curve.html#sphx-glr-auto-examples-calibration-plot-calibration-curve-py

Now I'm confused too: from the example, it seems to me that NB (with a transposed sigmoid shape) is 
**under** confident, while the LinearSVC (with a sigmoid shape) is **over**confident: for example for the bin at 0.8, its predictions are close to 1, so I interpret this as being over-confident about the positive class. What am I getting wrong? Maybe @ogrisel can chime in?

Reply:
> rom the example, it seems to me that NB (with a transposed sigmoid shape) is
> **under** confident, while the LinearSVC (with a sigmoid shape) is **over**confident:

Yes you are right! I should have thought about this more. (I think @ogrisel will be on holiday from next week though...)

Reply:
> What am I getting wrong?

Ok so what I was getting wrong is that I was inverting the axes: the x axis is what the classifiers predicts and the y axis are the actual proportions. So indeed the sigmoid curve describes under-confidence and the example is correct.

Reply:
Yes, okay. It is tricky to interpret. I guess you have to think about it from 0.5 and go up/down.